source/extensions/filters/http/grpc_http1_reverse_bridge/filter.cc

```diff
@@ -147,9 +151,30 @@ Http::FilterHeadersStatus Filter::encodeHeaders(Http::ResponseHeaderMap& headers
     headers.setContentType(content_type_);
 
     if (withhold_grpc_frames_) {
-      // Adjust content-length to account for the frame header that's added.
-      adjustContentLength(headers,
-                          [](auto length) { return length + Grpc::GRPC_FRAME_HEADER_SIZE; });
+      // If the upstream should set a header indicating the response size, use that.
+      if (response_size_header_) {
+        auto length_headers = headers.get(*response_size_header_);
+        if (!length_headers.empty() && absl::SimpleAtoi(length_headers[0]->value().getStringView(),
```

Comment:
FWIW this does not handle multiple inline headers 

so you'd parse
Length: 4
Length 4
as 4 but not
Length: 4, 4
which is also valid.  Is that intentional?

Reply:
I wrote in the API docs that only the first header would be used if multiple were present, since I can't think of a reasonable way to interpret repeated headers. Does that sound right to you? The other option I can think of would be to fail immediately, but that seems too strict to me.

It sounds like I need to read up on the spec a little bit, since I'm not familiar with repeated inline headers, but I plan to update the code to only use the first value in all cases.